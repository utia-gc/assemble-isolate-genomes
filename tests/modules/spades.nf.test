nextflow_process {

    name "Test Process spades"
    script "modules/spades.nf"
    process "spades"

    test("spades produces an assembly -- SE reads") {
        // instantiate helper objects
        def readsLane1 = new ReadsSELane1()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = [
                    [
                        sampleName:   "${readsLane1.getSampleName()}",
                        lane:         "${readsLane1.getLane()}",
                        readType:     "${readsLane1.getReadType()}",
                        trimStatus:   "raw",
                    ],
                    file("${readsLane1.getR1()}"),
                    file("${readsLane1.getR2()}")
                ]
                """
            }
        }

        then {
            // test process success
            assert process.success

            // test scaffolds
            with(process.out.scaffolds) {
                assert size() == 1
                with(get(0)) {
                    assert get(1) ==~ /^.*\/${readsLane1.getStemName()}\/scaffolds.fasta/
                }
            }
            // test contigs
            with(process.out.contigs) {
                assert size() == 1
                with(get(0)) {
                    assert get(1) ==~ /^.*\/${readsLane1.getStemName()}\/contigs.fasta/
                }
            }

            // test scaffolds and contigs reproducibility
            assert snapshot(process.out.scaffolds).match()
            assert snapshot(process.out.contigs).match()
        }

    }


    test("spades produces an assembly -- PE reads") {
        // instantiate helper objects
        def readsLane1 = new ReadsPELane1()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = [
                    [
                        sampleName:   "${readsLane1.getSampleName()}",
                        lane:         "${readsLane1.getLane()}",
                        readType:     "${readsLane1.getReadType()}",
                        trimStatus:   "raw",
                    ],
                    file("${readsLane1.getR1()}"),
                    file("${readsLane1.getR2()}")
                ]
                """
            }
        }

        then {
            // test process success
            assert process.success

            // test scaffolds
            with(process.out.scaffolds) {
                assert size() == 1
                with(get(0)) {
                    assert get(1) ==~ /^.*\/${readsLane1.getStemName()}\/scaffolds.fasta/
                }
            }
            // test contigs
            with(process.out.contigs) {
                assert size() == 1
                with(get(0)) {
                    assert get(1) ==~ /^.*\/${readsLane1.getStemName()}\/contigs.fasta/
                }
            }

            // test scaffolds and contigs reproducibility
            assert snapshot(process.out.scaffolds).match()
            assert snapshot(process.out.contigs).match()
        }

    }

}
