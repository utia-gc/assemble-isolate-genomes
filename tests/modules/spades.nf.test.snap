{
    "spades produces an assembly -- SE reads": {
        "content": [
            [
                [
                    {
                        "sampleName": "SRR1066657",
                        "lane": "001",
                        "readType": "single",
                        "trimStatus": "raw"
                    },
                    "contigs.fasta:md5,ed1435f7f97817068aa8f9a6e99e5a62"
                ]
            ]
        ],
        "timestamp": "2024-01-30T15:40:00.156024894"
    }
}